cognite/pygen/_core/templates/_api_client_multi_model.py.jinja

```diff
@@ -154,12 +163,19 @@ class {{ client_name }}:
         return self.upsert(items, replace, write_none)
 
     def delete(
-        self, external_id: str | SequenceNotStr[str], space: str = DEFAULT_INSTANCE_SPACE
+        self,
+        external_id: (
+            str | SequenceNotStr[str] | data_classes.DomainModelWrite | Sequence[data_classes.DomainModelWrite]
+        ),
+        space: str = DEFAULT_INSTANCE_SPACE
```

Comment:
Could it be an idea to add a depth parameter to specify the depth for which the delete operation should happen? 

We might not always want to delete the connecting node as well in the case where that node has edges to other nodes again, but we would always want to delete the edges (this is the scenario I describe in the comment to the test below).

So the default depth could be 0 - that upon deletion of a node, the edges going in and out of that node are deleted, and then you could provide depth=1, if you in addition want to delete the connecting nodes one level out, and possibly even 2 or 3?

Reply:
Hmm, would have to be connection hops or something as depth is a property of a tree not a graph. 

But I am slightly skeptical about it as it complicates the call. Even with a good default.